core/capabilities/reader.go

```diff
@@ -0,0 +1,104 @@
+package capabilities
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/keystone_capability_registry"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+	evmrelaytypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+type remoteRegistryReader struct {
+	r types.ContractReader
+}
+
+var _ reader = (*remoteRegistryReader)(nil)
+
+type capabilityID [32]byte
+
+type state struct {
+	DONs              []kcr.CapabilityRegistryDONInfo
+	IDsToNodes        map[p2ptypes.PeerID]kcr.CapabilityRegistryNodeInfo
+	IDsToCapabilities map[capabilityID]kcr.CapabilityRegistryCapability
+}
```

Comment:
Would it make sense to create mappings for all entities to keep things consistent?

```suggestion
type nodeID p2ptypes.PeerID
type capabilityID [32]byte
type donID uint32

type state struct {
	DONs                   map[donID]kcr.CapabilityRegistryDONInfo
	IDsToNodes         map[nodeID]kcr.CapabilityRegistryNodeInfo
	IDsToCapabilities map[capabilityID]kcr.CapabilityRegistryCapability
}
```

Reply:
also, I think `type capabilityID [32]byte` should be `type hashedCapabilityID [32]byte` for accuracy

Reply:
Yep, can do